apps/array/src/main/services/shell.ts

```diff
@@ -56,4 +57,23 @@ export function registerShellIpc(): void {
   handle("shell:get-process", (_event, sessionId: string) => {
     return shellManager.getProcess(sessionId);
   });
+
+  handle(
+    "shell:execute",
+    async (
+      _event,
+      cwd: string,
+      command: string,
+    ): Promise<{ stdout: string; stderr: string; exitCode: number }> => {
+      return new Promise((resolve) => {
+        exec(command, { cwd, timeout: 60000 }, (error, stdout, stderr) => {
```

Comment:
how do we feel about a timeout here? I worry if people are using this to run a dev server it might be an issue